src/Meziantou.Framework.NuGetPackageValidation/Internal/SharedHttpClient.cs

```diff
@@ -26,13 +26,26 @@ protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage
             const int MaxRetries = 5;
             for (var i = 1; ; i++)
             {
+                TimeSpan? delay = null;
                 HttpResponseMessage? result = null;
+
                 try
                 {
                     result = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
-                    if (!IsLastAttempt(i) && ((int)result.StatusCode >= 500 || result.StatusCode is System.Net.HttpStatusCode.RequestTimeout))
+                    if (!IsLastAttempt(i) && ((int)result.StatusCode >= 500 || result.StatusCode is System.Net.HttpStatusCode.RequestTimeout or System.Net.HttpStatusCode.TooManyRequests))
                     {
                         result.Dispose();
+
+                        // Use "Retry-After" value, if available. Typically, this is sent with
+                        // either a 503 (Service Unavailable) or 429 (Too Many Requests):
+                        // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Retry-After
+
+                        delay = result.Headers.RetryAfter switch
```

Comment:
I don't know if this is safe to access `result` right after disposing it. I think you should move the `Dispose` call after this one.

Reply:
> I don't know if this is safe to access `result` right after disposing it.

The [`Dispose` implementation of `HttpResponseMessage`](https://github.com/dotnet/runtime/blob/v7.0.8/src/libraries/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L212-L221) only [disposes the content](https://github.com/dotnet/runtime/blob/v7.0.8/src/libraries/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L219). It doesn't invalidate/reset the entire object's state. The [`Headers` getter makes no check about the disposed state of the object](https://github.com/dotnet/runtime/blob/v7.0.8/src/libraries/System.Net.Http/src/System/Net/Http/HttpResponseMessage.cs#L107). The [documentation](https://learn.microsoft.com/en-us/dotnet/api/system.net.http.httpresponsemessage.dispose) doesn't call out anything special either. That all said, I agree that it may be best not to rely on those implementation details and so I've moved the `Dispose` call after with 692eacbc4929a835efab87ca12497967e2ebe17d.